### ISSUE TITLE
Add use_count() method to resource_handle to return the number of references to the shared_ptr

### ISSUE THREAD

**Issue by GavinNL** [skypjack/entt#727]
Title is self explanatory.  

I think you mentioned you were going to change the implementation to use something other than shared_ptr, would having a reference count be feasible in the new implementation?

**Comment by skypjack** [comment]
> you mentioned you were going to change the implementation to use something other than shared_ptr

Yes and no. In theory, you can already do that, since the handle is fully customizable on user side.
In any case, if I'll change the default implementation in future, I don't plan to also break backward compatibility and it makes for a handle to have a counter of its copies, so... 🤷‍♂️
